src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -143,8 +143,8 @@ internal static bool TryGetDateTimeOffsetFromTimestampString(Dictionary<string,
         // Converts the specified DateTimeOffset to the string representation of seconds since the Unix Epoch.
         internal static string GetTimestampStringFromDateTimeOffset(DateTimeOffset timestamp)
         {
-            double secondsSinceEpoch = GetSecondsSinceEpochFromDateTimeOffset(timestamp);
-            return secondsSinceEpoch.ToString("F9", CultureInfo.InvariantCulture); // 6 decimals, no commas
+            decimal secondsSinceEpoch = GetSecondsSinceEpochFromDateTimeOffset(timestamp);
+            return secondsSinceEpoch.ToString("F9", CultureInfo.InvariantCulture); // 9 decimals, no commas
```

Comment:
Don't we want `G9`?

![image](https://user-images.githubusercontent.com/8291187/174835319-00f5ab2c-c8b7-4333-a612-abd16508d34b.png)

Reply:
Hmm, maybe `9` is too small.

![image](https://user-images.githubusercontent.com/8291187/174836196-8bd4b907-0740-4f24-9f4d-927495aa815f.png)

Reply:
Maybe we just want `G`...

https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#general-format-specifier-g

Reply:
Just `G` (the default) will print the shortest roundtrippable string. `G9` will print the shortest string or up to `9` significant digits, whichever is lesser. However, it will also print exponentials.

There is actually a decent amount of "inconsistency" in how the formatting APIs work between the different format specifiers and its a bit frustrating at times, but we can't really change existing ones due to back-compat.

Reply:
> However, it will also print exponentials.

I had initially tested `G` and because it shows `E+XX` like in the screenshot above, I decided to use `F` instead.

@tannergooding just to make sure we are all in agreement, using `F` is ok?

Reply:
It won't print `E+XX` if you just use `G`

```C#
return secondsSinceEpoch.ToString("G", CultureInfo.InvariantCulture);
```

Reply:
> It won't print E+XX if you just use G

It can. `G` picks the more compact of `F` or `E`, but without including trailing non-significant digits.

Reply:
Gah, nevermind. This is a case where just `G` for `decimal` always picks `F` (basically, and therefore differentiating from the docs/convention), but `G##` picks `F` or `E` an.

I refer back to my earlier statement
> There is actually a decent amount of "inconsistency"

Reply:
> In what scenarios?

That table is quite a bit out of date and doesn't exactly line up with what the implementations do. `Double` and `Single` are just incorrect here.

Reply:
> There is actually a decent amount of "inconsistency"

Totally agree. I have to look it up every time.

> That table is quite a bit out of date and doesn't exactly line up with what the implementations do. Double and Single are just incorrect here.

😢 . Can we get it updated? I think people really rely on this kind of information (like I am above).

Reply:
https://github.com/dotnet/docs/issues/29952 has the full details and repro steps.

Reply:
> In what scenarios?

```cs

double d = 1655815010.19999999998;
Console.WriteLine(d.ToString("G9", CultureInfo.InvariantCulture)); // 1.65581501E+09
d = 1.19999999998;
Console.WriteLine(d.ToString("G9", CultureInfo.InvariantCulture)); // 1.2
d = 123456789.19999999998;
Console.WriteLine(d.ToString("G9", CultureInfo.InvariantCulture)); // 123456789   (lol wat?)
```

Reply:
We are using a `decimal` in the code.

Reply:
With decimal, `F9` works fine:

```cs
decimal d = 1655815010.199999998M;
Console.WriteLine(d.ToString("F9", CultureInfo.InvariantCulture)); // 1655815010.199999998
d = 1.199999998M;
Console.WriteLine(d.ToString("F9", CultureInfo.InvariantCulture)); // 1.199999998
d = 123456789.199999998M;
Console.WriteLine(d.ToString("F9", CultureInfo.InvariantCulture)); // 123456789.199999998
```

But `G9` behaves similarly to the example above for double:

```cs
decimal d = 1655815010.199999998M;
Console.WriteLine(d.ToString("G9", CultureInfo.InvariantCulture)); //1.65581501E+09
d = 1.199999998M;
Console.WriteLine(d.ToString("G9", CultureInfo.InvariantCulture)); // 1.2
d = 123456789.199999998M;
Console.WriteLine(d.ToString("G9", CultureInfo.InvariantCulture)); // 123456789
```

Reply:
I'll use `G` as suggested above, and as explained via chat.

I am also going to remove some calls to this conversion code where it's not necessary. This code to convert timestamps is only required when converting from gnu to pax or viceversa (to store atime and ctime as required by the format).